src/main/java/com/ning/http/client/Realm.java

```diff
@@ -205,6 +208,15 @@ public boolean isTargetProxy() {
         return targetProxy;
     }
 
+    /**
+     * Return true if the provided credentials may vary in some future request.
+     *
+     * @return true if the provided credentials may vary in some future request.
+     */
+    public boolean credentialMayVary() {
```

Comment:
I would use a more generic name to declare what it effectively does, especially on the grizzly level. For example, forConnectionClosed.